Modules/Core/Common/include/itkMacro.h

```diff
@@ -610,15 +610,8 @@ OutputWindowDisplayDebugText(const char *);
 //   Provide legacy methods with no warnings.
 #    define itkLegacyMacro(method) method
 #  else
-//   Setup compile-time warnings for uses of deprecated methods if
-//   possible on this compiler.
-#    if defined(__GNUC__) && !defined(__INTEL_COMPILER)
-#      define itkLegacyMacro(method) method __attribute__((deprecated))
-#    elif defined(_MSC_VER)
-#      define itkLegacyMacro(method) __declspec(deprecated) method
-#    else
-#      define itkLegacyMacro(method) method
-#    endif
+//   Request compile-time warnings for uses of deprecated methods.
+#    define itkLegacyMacro(method) [[deprecated]] method
```

Comment:
You could even consider to add a message, `[[deprecated("reason")]]`, as I see at https://en.cppreference.com/w/cpp/language/attributes/deprecated  I think it might be useful for users to explicitly mention `ITK_LEGACY_REMOVE` in the message, for example:

     [[deprecated("This is a legacy method that may be removed by ITK_LEGACY_REMOVE")]]

Reply:
This is a warning which will appear for library users. And instructions how to remove the method they are using would not be too helpful.

Reply:
> This is a warning which will appear for library users. And instructions how to remove the method they are using would not be too helpful.

OK, but do you have a better suggestion?

    [[deprecated("This is a legacy method that is only available as long as ITK_LEGACY_REMOVE=OFF")]]

?

Reply:
That is better. Maybe also write that it will be removed with next major version of ITK? As this PR was merged, it will need to go into a follow-up.

Reply:
@dzenanz OK, what about the following? 

  "This is a legacy method. It is only available when ITK_LEGACY_REMOVE=OFF, and it may be removed with the next major ITK release"

I chose "may be removed", instead of "will be removed", as you never know for sure, right? 😺. Or would you prefer, "it is intended to be removed"?

Reply:
> intended to be removed

This is the truth! Still leaves some wiggle room in case we decide to change mind. I like this the best.